src/Microsoft.AspNet.Razor/TagHelpers/TagMode.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.TagHelpers
```

Comment:
Thoughts on throwing this in the `Microsoft.AspNet.Razor.Runtime.TagHelpers` namespace despite its location? That way when writing your `TagHelper`s/`TargetElement`s you wont need to bring in any additional usings. Will also make codegen simpler.